crypto/threads_pthread.c

```diff
@@ -408,6 +408,13 @@ static struct rcu_qp *update_qp(CRYPTO_RCU_LOCK *lock, uint32_t *curr_id)
     ATOMIC_STORE_N(uint32_t, &lock->reader_idx, lock->current_alloc_idx,
                    __ATOMIC_RELAXED);
 
+    /*
```

Comment:
I'm sure I'm missing something. in my opinion we should be doing __ATOMIC_RELEASE on lin 409, because `update_qp()` reads from `&lock->reader_idx` before it modifies `->qp_groups[->reader_idx]`

Reply:
no that is completely irrelevant, the algorithm would work even if the qp_idx was a random value